### PR TITLE
'Open folder' button

### PR DESCRIPTION
**Pull request by diegous**
Changed button to load entire folder instead of single file. Adjusted regular expressions to accept all files in sample.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -42,12 +41,38 @@
...
+
     loadButton.onclick = function () {
-      filePathElement = document.getElementById('filePath')
-      filePath = filePathElement.files[0].path
-      fileName = filePathElement.files[0].name
-      loadFile(filePath, fileName)
+      dialog.showOpenDialog({ properties: ['openDirectory'] }, function (paths) {
+        let dirPath = paths[0]
+        let encoding = 'CP437'
+        let Iconv = require('iconv').Iconv;
```
Comment (diegous) [inline index.html]: This is already declared in `main.js` so this might be redundant.

File: js/fileMatchers.js [hunk 2]
```diff
@@ -4,22 +4,22 @@ const headerSubtitleRegex =           /│           Dr. Enrique C. Muñoa
...
 const headerProtocolRegex =           /├───────── Teléfono: \(0943\) 74373 ────────┘\s*Protocolo N°\s*(\d+)\s*│/
 const headerBlankRegex =              /│                                                                    │/
-const headerPatientDataRegex =        /│Paciente: (.+) Edad: (\d+)\s*│/
+const headerPatientDataRegex =        /│Paciente:(.+)Edad:(.+)\s*│/
 //headerBlankRegex                    /│                                                                    │/
-const headerDoctorAndInsuranceRegex = /│Solicitado por (.+)\s+O. Social: (.+) │/
+const headerDoctorAndInsuranceRegex = /│Solicitado por (.+)O. Social:(.+)│/
 const headerBottomBorderRegex =       /└────────────────────────────────────────────────────────────────────┘/
-const materialRegex =                 /Material (?:R|r)emitido:(.+)/
+const materialRegex =                 /Material (?:R|r)emitido\s*:(.+)/
```
Comment (diegous) [inline js/fileMatchers.js]: This regex might be benefited by adding `gi` at the end (just like `descriptionRegex`) to make it case insensitive.

File: package.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
   "dependencies": {
     "dexie": "^1.5.1",
     "electron": "^1.4.1",
+    "iconv": "^2.2.3",
     "knex": "^0.12.8",
     "lodash": "^4.17.2",
     "sqlite3": "^3.1.8"
```
Comment (diegous) [inline package.json]: `sqlite` will not be used, so it should be removed.

File: index.html [hunk 4]
```diff
@@ -42,12 +41,38 @@
 
   <script>
 
+    const fs = require('fs')
+    const dialog = require('electron').remote.dialog
```
Comment (diegous) [inline index.html]: Maybe it is better to declare these in `main.js`? I have no idea

File: index.html (hunk 1 above)
Comment (betomoretti) [inline index.html]: Check this docs: [Join paths](https://nodejs.org/api/path.html#path_path_join_paths)

File: index.html [hunk 5]
```diff
@@ -42,12 +41,38 @@
...
+        let encoding = 'CP437'
+        let Iconv = require('iconv').Iconv;
+        let iconv = new Iconv(encoding, 'UTF-8')
+
+        fs.readdir(dirPath, function (err, files) {
+          if (err) return;
+
+          files.forEach( function (fileName) {
+            // I don't know if this '/' will work on Window$ since it uses \ for dirs
+            let filePath = dirPath +'/'+ fileName
```
Comment (betomoretti) [inline index.html]: Check this docs: [Join paths](https://nodejs.org/api/path.html#path_path_join_paths)

File: js/fileLoader.js [hunk 6]
```diff
@@ -94,7 +90,7 @@ class BodyMatcher {
 
     // If all failed, its a pargraph
     if (!match)
-      return { match: null }
+      return { match: null, regex: null }
```
Comment (betomoretti) [inline js/fileLoader.js]: It is not necessary return null value for empty keys. You can just return empty object so you can check for obj.match and will be undefined. Used in a condition can be considered as falsy.

**Pull request merged**